### ISSUE TITLE
Interpreting first.label vs. label?

### ISSUE THREAD

**Issue by bbimber** [LTLA/SingleR#26]
Hello,

The dataframe returned by SingleR has 'label', first.label and score.  I find instances where cells have the highest score as 'NK_cell', label=T_cells and first.label=NK_cell.  this is being generated using hpca as the reference.  Do you have a suggestion on how I should interpret this?

Related to that: do you expect that users might interrogate the score matrix and set some kind of threshold, or are you assuming SingleR has internally done this and one would interpret any call to be of good quality?

**Comment by LTLA** [comment]
If you want to combine references, I just added a `matchReferences` function (see #25) that should help in matching up labels across two sets of references. Not quite fully automatic but it should get you pretty close. I wrote it with single-cell references in mind as the statistics are only interesting if you have lots of representatives per label; but it _should_ work okay with bulk references, it would just basically be looking for mutual nearest neighbors in that application.
